williballenthin/viv-utils
# Prep v0.7

i've been working over here but might as well put it all in master

@mr-tz PR here as an FYI

adds:
  - tests
  - mypy
  - black, sort

fixes:
  - various APIs that didn't work as we thought

changes:
  - py3 only
  - remove emu arg from monitor ctor
  - make hooks simply a callable, like a function or class with `__call__`. no longer a subclass of `Hook`
  - snake case method names
  - make dbg driver breakpoint API via property access, not `addBreakpoint`
  - dbg driver `run_to_*` raises exceptions more reliably, probably need to handle them in client code
